parley/src/icu_convert.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2024 the Parley Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use crate::analysis::AnalysisDataSources;
+
+use icu_locale_core::LanguageIdentifier;
+use icu_properties::props::Script;
+
+pub(crate) fn script_to_fontique(
+    script: Script,
+    analysis_data_sources: &AnalysisDataSources,
+) -> fontique::Script {
+    let short_name: [u8; 4] = analysis_data_sources
+        .script_short_name()
+        .get(script)
+        .unwrap_or("Zzzz")
+        .as_bytes()
+        .try_into()
+        .expect("exactly 4 bytes");
```

Comment:
you can get an `icu::locale::subtags::Script` from the `icu::properties::props::Script`:

```suggestion
        .get_locale_script(script)
        .unwrap_or(icu_locale_core::subtags::script!("Zzzz"))
        .into_raw();
```